src/Microsoft.NetCore.Analyzers/UnitTests/Tasks/UseValueTaskCorrectlyTests.cs

```diff
@@ -0,0 +1,1274 @@
+                        Helpers.ReturnsValueTask().AsTask();
+
+                        new ValueTask<string>(""hello"").AsTask();
+                        new ValueTask<string>(Task.FromResult(""hello"")).AsTask();
+                        Helpers.ReturnsValueTaskOfT<string>().AsTask();
+
+                        Helpers.ReturnsValueTaskOfInt().AsTask();
+                    }
+                }"), parseOptions: s_csharpPreviewParseOptions);
+        }
+
+        [Fact]
+        public void NoDiagnostics_ReturnValueTask()
+        {
+            VerifyCSharp(CSharpBoilerplate(@"
+                using System;
+                using System.Threading.Tasks;
+
+                class C
+                {
+                    public ValueTask ReturnValueTaskExpression() => Helpers.ReturnsValueTask();
+
+                    public ValueTask<T> ReturnValueTaskOfTExpression<T>() => Helpers.ReturnsValueTaskOfT<T>();
+
+                    public ValueTask<int> ReturnValueTaskOfIntExpression() => Helpers.ReturnsValueTaskOfInt();
+
+                    public ValueTask ReturnValueTask()
+                    {
+                        Console.WriteLine();
+                        return Helpers.ReturnsValueTask();
+                    }
+
+                    public ValueTask<T> ReturnValueTaskOfT<T>()
+                    {
+                        Console.WriteLine();
+                        return Helpers.ReturnsValueTaskOfT<T>();
+                    }
+
+                    public ValueTask<int> ReturnValueTaskOfInt()
+                    {
+                        Console.WriteLine();
+                        return Helpers.ReturnsValueTaskOfInt();
+                    }
+
+                }"), parseOptions: s_csharpPreviewParseOptions);
+        }
+
+        [Fact]
+        public void NoDiagnostics_OutValueTask()
+        {
+            VerifyCSharp(CSharpBoilerplate(@"
+                using System;
+                using System.Threading.Tasks;
+
+                class C
+                {
+                    public void OutValueTask(out ValueTask vt)
+                    {
+                        vt = Helpers.ReturnsValueTask();
+                    }
+
+                    public void OutValueTaskOfT<T>(out ValueTask<T> vt)
+                    {
+                        vt = Helpers.ReturnsValueTaskOfT<T>();
+                    }
+
+                    public void OutValueTaskOfInt(out ValueTask<int> vt)
+                    {
+                        vt = Helpers.ReturnsValueTaskOfInt();
+                    }
+                }"), parseOptions: s_csharpPreviewParseOptions);
+        }
+
+        [Fact]
+        public void NoDiagnostics_WorkBetweenStoreAndAwait()
+        {
+            VerifyCSharp(CSharpBoilerplate(@"
+                using System;
+                using System.Threading.Tasks;
+
+                class C
+                {
+                    public async Task WorkBetweenStoreAndAwait()
```

Comment:
Can you add unit tests where task is created in a `try` block and awaited in `finally` and/or `catch` blocks?